### PR TITLE
#13264: Update Backward Relational ops

### LINKED ISSUE #13264: Update Relational ops
In #10543 , All of the overloaded comparison ops has been removed . It was found that 5 comparison ops (except for `LT_BW`) is required. So , Support for those ops have is required

- [x] eq_bw
- [x] ge_bw
- [x] gt_bw
- [x] le_bw
- [x] ne_bw

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Issue : #13264

**What problem you solve :**  
       In #10543 , All of the overloaded comparison ops has been removed . It was found that 5 comparison ops (except for LT_BW) is required. 

**What changed** : 
      Support for 5 relational ops

https://github.com/tenstorrent/tt-metal/actions/runs/11159646565 - passed

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/backward/test_backward_ ne.py
Comment (KalaivaniMCW) [inline tests/ttnn/unit_tests/operations/backward/test_backward_ ne.py:1]: the filename has an extra space here _ne

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp [hunk 1]
```diff
@@ -198,6 +198,39 @@ std::vector<ComplexTensor> ExecuteBackwardAdd::invoke(
...
+    result[0] = input_grad.has_value() ? ttnn::zeros_like(queue_id, grad, grad.get_dtype(), grad.get_layout(), std::nullopt, output_mem_config, input_grad) :  ttnn::zeros_like(queue_id, grad, grad.get_dtype(), grad.get_layout(), std::nullopt, output_mem_config);
+    return result;
+}
+
+std::vector<std::optional<Tensor>> ExecuteBackwardEQ::invoke(
+    const Tensor& grad, const Tensor& input, const Tensor& other, const std::optional<MemoryConfig>& output_mem_config,
+    const std::vector<bool>& are_required_outputs,
+    std::optional<Tensor> input_grad,
+    std::optional<Tensor> other_grad) {
+    return ExecuteBackwardLT::invoke(ttnn::DefaultQueueId, grad, input, other, output_mem_config, are_required_outputs, input_grad, other_grad);
```
Comment (KalaivaniMCW) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:225]: ExecuteBackwardLT::invoke ? EQ?

Comment (VirdhatchaniKN) [reply ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:225]: I'll Update the code

**Review by Aswinmcw — APPROVED** [review]
LGTM

File: ttnn/ttnn/operations/binary_backward.py [hunk 2]
```diff
@@ -186,6 +186,41 @@ def _golden_function_comparison_ops(torch_op, grad_tensor, input_tensor_a, input
     ),
 )
 
+ttnn.attach_golden_function(
+    ttnn.eq_bw,
+    golden_function=lambda grad, a, b, *args, **kwargs: _golden_function_comparison_ops(
+        torch.eq, grad, a, b, *args, **kwargs
```
Comment (eyonland) [inline ttnn/ttnn/operations/binary_backward.py:192]: I don't follow what is going on here.

**Review by KalaivaniMCW — APPROVED** [review]
LGTM

**Review by eyonland — APPROVED** [review]

File: ttnn/ttnn/operations/binary_backward.py (hunk 2 above)
Comment (eyonland) [inline ttnn/ttnn/operations/binary_backward.py:192]: Given the these comparison ops are not differentiable and always end up with zero tensors, it does make sense to make this common code as you mentioned earlier.

**Review by eyonland — APPROVED** [review]

**Review by mouliraj-mcw — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/11159646565 - passed

**Review by umadevimcw — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
Closing PR as Op requirement is not needed

**Pull request closed without merging**